### PR TITLE
Describe data format

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: docs/fobnail-data-format.md
Comment (TomaszAIR) [inline docs/fobnail-data-format.md:29]: @ arturkow2000 can we describe each key that is part of this structure?

File: docs/fobnail-data-format.md
Comment (TomaszAIR) [inline docs/fobnail-data-format.md:51]: @arturkow2000 also here, some description of the JSON keys may be helpful, e.g. I do not see explaining for `update_ctr`

File: docs/fobnail-data-format.md
Comment (TomaszAIR) [inline docs/fobnail-data-format.md:42]: @arturkow2000 so RIMs are also sent in CBOR format (like metadata?)

**Review by TomaszAIR — CHANGES_REQUESTED** [review]

File: docs/fobnail-data-format.md
Comment (krystian-hebel) [reply docs/fobnail-data-format.md:29]: :+1: also CBOR major type of each field would be good. It would be nice to also mention where this data comes from, but perhaps this should be in `metadata.md` instead.

File: docs/fobnail-data-format.md
Comment (krystian-hebel) [inline docs/fobnail-data-format.md:46]: Drop MSB from description. CBOR major type by itself doesn't differentiate between different sizes of integer values so it doesn't have to be always 32 bits (in fact, it probably is sent in 3 bytes in current form)
```suggestion
of PCR itself. The LSB of bitmap corresponds to PCR0.
```

File: docs/fobnail-data-format.md
Comment (krystian-hebel) [inline docs/fobnail-data-format.md:57]: Maybe add a comment that there are as many elements as set bits in `pcrs`?

File: docs/fobnail-data-format.md
Comment (krystian-hebel) [inline docs/fobnail-data-format.md:83]: That doesn't tell us where exactly nonce is added (prepended vs. appended), nor whether it is added to CBOR or as plain bytes before/after CBOR.

File: docs/fobnail-data-format.md
Comment (arturkow2000) [reply docs/fobnail-data-format.md:30]: Added brief description of each field including major types.
Details (like how to determine primary network adapter) are described in metadata.md which is linked here

File: docs/fobnail-data-format.md
Comment (arturkow2000) [reply docs/fobnail-data-format.md:56]: Added explanation

File: docs/fobnail-data-format.md
Comment (arturkow2000) [reply docs/fobnail-data-format.md:47]: Yes, I noted this in document

File: docs/fobnail-data-format.md
Comment (arturkow2000) [reply docs/fobnail-data-format.md]: Done

File: docs/fobnail-data-format.md
Comment (arturkow2000) [reply docs/fobnail-data-format.md:99]: Fixed

File: docs/fobnail-data-format.md
Comment (krystian-hebel) [inline docs/fobnail-data-format.md:45]: I think we should redefine RIM format:

* similar but stripped (no `update_ctr` and `pcr` fields) format will be used for appraisal policy
* for appraisal policy order of hash algorithms matters, this is not possible to encode in current format
* there may appear additional hash algorithms for different TPMs

My suggestion is to switch from one map per hash (with map name = hash algorithm name) to array of maps (with algorithm name as key/value pair inside each map). We could also add a field specifying size of given hash, but I don't think it is necessary, it should be enough to test if all hashes in bank have the same length.

Comment (krystian-hebel) [reply docs/fobnail-data-format.md:45]: @arturkow2000 is it feasible to do with current Fobnail code?

**Review by krystian-hebel — APPROVED** [review]
Approved, but lets merge it after changes to code are merged.

File: docs/fobnail-data-format.md
Comment (krystian-hebel) [inline docs/fobnail-data-format.md:62]: So, in case of "new" algorithms, name may be not known either by Fobnail, Attester or both. It would be safer to pass algorithm ID as an integer.

Also, comment above this field doesn't specify CBOR major type.

**Review by krystian-hebel — CHANGES_REQUESTED** [review]
Some issues became apparent after testing code that require additional changes to format of data.

**Pull request merged**
